// MTRX3760 2025 Project 2: Battery Terminal Display
// File: battery_terminal_display.hpp
//
// PURPOSE: Display battery information in terminal with color coding
// - Subscribes to /battery_state topic
// - Shows voltage, percentage, time remaining
// - Color-coded warnings (green/yellow/red)
// - Real-time updates in terminal
//
// TODO: Implement terminal color output (ANSI codes)
// TODO: Add battery state subscriber
// TODO: Format display output

#ifndef BATTERY_TERMINAL_DISPLAY_HPP
#define BATTERY_TERMINAL_DISPLAY_HPP

// Add includes here

namespace battery {

class BatteryTerminalDisplay {
public:
    // Constructor/Destructor
    
    // Display update functions
    
    // Color formatting helpers
    
private:
    // ROS node handle
    
    // Subscribers
    
    // Display formatting
    
    // Battery state callback
};

} // namespace battery

#endif // BATTERY_TERMINAL_DISPLAY_HPP
